packages/storage/src/cookies.ts

```diff
@@ -29,22 +29,15 @@ const cookiePropertyKeys = [
 ] as const;
 
 function serializeCookieOptions(options?: CookieOptions) {
-  if (!options) {
-    return "";
-  }
-  let memo = "";
-  for (const key in options) {
-    if (!cookiePropertyKeys.includes(key as keyof CookieProperties)) continue;
-
-    const value = options[key as keyof CookieProperties];
-    memo +=
-      value instanceof Date
-        ? `; ${key}=${value.toUTCString()}`
-        : typeof value === "boolean"
-          ? `; ${key}`
-          : `; ${key}=${value}`;
-  }
-  return memo;
+  if (!options) return "";
+  return Object.entries(options).map(([key, value]) => {
+    key = key.replace(/([a-z])([A-Z])/g, (_, a, b) => `${a}-${b.toLowerCase()}`);
```

Comment:
This breaks httpOnly and sameSite? which is not hyphenated?

I think only max-age is hyphenated